tests/unit/test_allocator.cpp

```diff
@@ -0,0 +1,182 @@
+    allocations += 2 * sizeof(VectorBlockMember *); // resize idToVectorBlockMemberMapping to 4
+    allocations += sizeof(VectorBlock);             // New vector block
+    allocations += sizeof(VectorBlockMember);
+    allocations += sizeof(VectorBlockMember *);          // Pointer for the new vector block member
+    allocations += sizeof(float) * dim;                  // keep the vector in the vector block
+    allocations += sizeof(VectorBlock *);                // Keep the allocated vector block
+    allocations += sizeof(std::pair<labelType, idType>); // keep the mapping
+    ASSERT_GE(allocator->getAllocationSize(), allocations);
+    info = bfIndex->info();
+    ASSERT_EQ(allocator->getAllocationSize(), info.memory);
+
+    bfIndex->deleteVector(2);
+    allocations -= sizeof(VectorBlock); // New vector block
+    allocations -= sizeof(VectorBlockMember);
+    allocations -= sizeof(VectorBlockMember *); // Pointer for the new vector block member
+    allocations -= sizeof(float) * dim;         // keep the vector in the vector block
+    ASSERT_GE(allocator->getAllocationSize(), allocations);
+    info = bfIndex->info();
+    ASSERT_EQ(allocator->getAllocationSize(), info.memory);
+
+    bfIndex->deleteVector(1);
+    allocations -= sizeof(VectorBlock); // New vector block
+    allocations -= sizeof(VectorBlockMember);
+    allocations -= sizeof(VectorBlockMember *); // Pointer for the new vector block member
+    allocations -= sizeof(float) * dim;         // keep the vector in the vector block
+    ASSERT_GE(allocator->getAllocationSize(), allocations);
+    info = bfIndex->info();
+    ASSERT_EQ(allocator->getAllocationSize(), info.memory);
+    delete bfIndex;
+}
+
+TEST_F(AllocatorTest, test_hnsw) {
+    std::shared_ptr<VecSimAllocator> allocator = std::make_shared<VecSimAllocator>();
+    size_t d = 128;
+
+    // Build with default args
+    VecSimParams params = {
+        .hnswParams =
+            {
+                .initialCapacity = 0,
+            },
+        .type = VecSimType_FLOAT32,
+        .size = d,
+        .metric = VecSimMetric_L2,
+        .algo = VecSimAlgo_HNSWLIB,
+    };
+
+    float vec[128] = {};
+    HNSWIndex *hnswIndex = new (allocator) HNSWIndex(&params, allocator);
+    ASSERT_GE(allocator->getAllocationSize(), sizeof(VecSimAllocator) + sizeof(HNSWIndex));
+    VecSimIndexInfo info = hnswIndex->info();
+    ASSERT_EQ(allocator->getAllocationSize(), info.memory);
+    size_t current_memory = info.memory;
+
+    hnswIndex->addVector(vec, 1);
+    ASSERT_GE(allocator->getAllocationSize(), current_memory);
+    info = hnswIndex->info();
+    ASSERT_EQ(allocator->getAllocationSize(), info.memory);
+    current_memory = info.memory;
+
+    hnswIndex->addVector(vec, 2);
+    ASSERT_GE(allocator->getAllocationSize(), current_memory);
+    info = hnswIndex->info();
+    ASSERT_EQ(allocator->getAllocationSize(), info.memory);
+
+    // TODO: commented out since hnsw does not recalim memory
```

Comment:
What do you mean by "hnsw does not reclaim memory"?

Reply:
when you delete a vector from hnsw, not all the memory that was allocated per this vector is reclaimed